src/data/sparse_page_source.cc

```diff
@@ -8,11 +8,9 @@
 #include <memory>
 #include <vector>
 #include <string>
-#include <locale>
```

Comment:
@rongou You need to add this header back in so that we can use this function: https://en.cppreference.com/w/cpp/locale/isalpha

Reply:
Thanks! CLion too eager removing unused includes.

Reply:
It's because the header is only used for Windows target.